dataobj/schedule.cc

```diff
@@ -669,12 +704,26 @@ void schedule_t::gimme_stop_name(cbuffer_t& buf, karte_t* welt, player_t const*
 	}
 }
 
-schedule_entry_t const& schedule_t::get_next_entry() const {
+schedule_entry_t const& schedule_t::get_next_entry() {
 	if(  entries.empty()  ) {
 		return dummy_entry;
-	} else {
-		return entries[(current_stop+1)%entries.get_count()];
+	} 
+	if(  check_next_line_valid()  &&  current_stop==entries.get_count()-1  ){
+		return next_line->get_schedule()->at(1);
+	}
```

Comment:
indexを1にするのは自明ではないので、コメントを足しておきましょう。
```suggestion
	if(  check_next_line_valid()  &&  current_stop==entries.get_count()-1  ){
		// Use the index 1, because the index 0 is same as the last entry of this schedule.
		return next_line->get_schedule()->at(1);
	}
```